scripts/deploy_protocol.py

```diff
@@ -1,39 +1,55 @@
-import sys
-import asyncio
+import os
+import logging
+import json
+from asyncio import run
 from starknet_py.net.account.account_client import (AccountClient)
 from starknet_py.net.signer.stark_curve_signer import KeyPair
 from starknet_py.contract import Contract
 from starknet_py.net.models import StarknetChainId
 from starkware.crypto.signature.signature import private_to_stark_key
 from starknet_py.net.gateway_client import GatewayClient
-from utils import deployContract
+from utils import declare_and_deploy_contract, declare_contract
 from pathlib import Path
+from dotenv import load_dotenv
+
+#Logging conf, just want to print results to console
+logging.basicConfig(level = logging.INFO)
+
+# Address of the ETH token on mainnet,testnet1 and testnet2
+ETH_ADDRESS = 2087021424722619777119509474943472645767659996348769578120564519014510906823
+# Set a high max fee for the deployments
+MAX_FEE = int(1e16)
+
+# Loading .env file
+dotenv_path = os.path.join('./', '.env')
+load_dotenv(dotenv_path)
+
+# Get env variables
+private_key = int(os.environ.get("PRIVATE_KEY"))
+account_address = int(os.environ.get("ACCOUNT_ADDRESS"),16)
+network = os.getenv('NETWORK')
+
+#Current starknet.py version does not support testnet2 as a default network
+if network == "testnet2":
+    network = "https://alpha4-2.starknet.io"
 
 #Configure Admin AccountClient
-private_key = int(sys.argv[1])
-account_address = int(sys.argv[2],16)
 public_key = private_to_stark_key(private_key)
 signer_key_pair = KeyPair(private_key,public_key)
-client = AccountClient(address=account_address, client=GatewayClient(net="testnet"), key_pair=signer_key_pair, chain=StarknetChainId.TESTNET, supported_tx_version=1)
-kakarot_abi = [
-    {
-        "inputs": [
-            {"name": "registry_address_", "type": "felt"},
-        ],
-        "name": "set_account_registry",
-        "outputs": [],
-        "type": "function",
-    }
-]
-
-async def deployKakarot():
-
-    print("----------------------------------")
-    print("--- Deploying Kakarot Protocol ---")
-    print("----------------------------------")
-    print(".")
-    print(".")
-    print(".")
+client = AccountClient(address=account_address, client=GatewayClient(net=network), key_pair=signer_key_pair, chain=StarknetChainId.TESTNET, supported_tx_version=1)
+#Get Kakarot ABI
+file = open("build/kakarot_abi.json")
```

Comment:
```
with open(...):
    json.load(...)
```
is preferred.

I'm wondering is `json.load(open(""))` keeps the file open or is good as well